qiskit/transpiler/passes/routing/sabre_swap.py

```diff
@@ -148,10 +148,11 @@ def __init__(self, coupling_map, heuristic="basic", seed=None, fake_run=False, t
             self.coupling_map = coupling_map
             self.target = None
         if self.coupling_map is not None and not self.coupling_map.is_symmetric:
-            # A deepcopy is needed here to avoid modifications updating
-            # shared references in passes which require directional
-            # constraints
-            self.coupling_map = deepcopy(self.coupling_map)
+            # A deepcopy is needed here if we don't own the coupling map (i.e. we were given it,
+            # rather than calculated it from the Target), to avoid modifications updating shared
+            # references in passes which require directional constraints.
+            if self.target is None:
```

Comment:
Just to clarify, if target is not `None`, then we're the owner of the coupling map since we'd have used the target to generate our own?

Reply:
yeah, after this change, `build_coupling_map` always returns an object that's fully owned by the caller, and we were the caller